samples/boards/stm32/power_mgmt/wkup_pins_gpio/src/main.c

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) 2024 STMicroelectronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/drivers/hwinfo.h>
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/sys/printk.h>
+#include <zephyr/sys/poweroff.h>
+#include <zephyr/sys/wakeup_source.h>
+
+#define STACKSIZE     1024
+#define PRIORITY      7
+#define WAIT_TIME_US (1 << 22)
+
+#define SW_WKUP_NODE DT_ALIAS(sw_wkup)
+#if !DT_NODE_HAS_STATUS(SW_WKUP_NODE, okay)
+#error "Unsupported board: sw_wkup devicetree alias is not defined"
+#endif
+
+static const struct gpio_dt_spec button = GPIO_DT_SPEC_GET_OR(SW_WKUP_NODE, gpios, {0});
+
+static const struct gpio_dt_spec led = GPIO_DT_SPEC_GET(DT_ALIAS(led0), gpios);
+
+static const struct wakeup_source *user_button_ws = WAKEUP_SOURCE_DT_GET(SW_WKUP_NODE);
+
+int main(void)
+{
+	int ret;
+	uint32_t cause;
+
+	hwinfo_get_reset_cause(&cause);
+	hwinfo_clear_reset_cause();
+
+	if (cause == (RESET_PIN | RESET_BROWNOUT)) {
+		printk("\nReset cause: Shutdown mode or power up\n\n");
+	}
+
+	if (cause == RESET_PIN) {
+		printk("\nReset cause: Reset pin\n\n");
+	}
+
+	__ASSERT_NO_MSG(gpio_is_ready_dt(&led));
+	if (!gpio_is_ready_dt(&button)) {
+		printk("Error: button device %s is not ready\n", button.port->name);
+		return 0;
+	}
+
+	ret = gpio_pin_configure_dt(&button, GPIO_INPUT);
```

Comment:
Why the below not fulfill the requirement? Or what is the problem with below?

```diff
-	ret = gpio_pin_configure_dt(&button, GPIO_INPUT);
+	ret = gpio_pin_configure_dt(&button, GPIO_INPUT | GPIO_WAKEUP);
```

Reply:
The goal of this PR is to verify that the wake-up source subsystem, proposed in https://github.com/zephyrproject-rtos/zephyr/pull/64814, is compatible with the way wake-up pins are configured on STM32 SoCs, no matter the wake-up source. So the implementation is not meant to be specific to GPIOs, even though we only implemented the z_sys_wakeup_source_enable_gpio(), since GPIOs are just one wake-up source among many for STM32 wake-up pins (section 10.3 Table 97 https://www.st.com/resource/en/reference_manual/rm0456-stm32u5-series-armbased-32bit-mcus-stmicroelectronics.pdf). Besides, STM32 wake-up pins registers are part of the Power Control peripheral & so it would not be right, in my opinion, to involve the GPIO or GPIO-keys drivers in their configuration.